olp-cpp-sdk-core/src/context/Context.cpp

```diff
@@ -148,46 +127,84 @@ void Context::init(JavaVM* vm, jobject context) {
 }
 
 JavaVM* Context::getJavaVM() {
-  auto cd = instance();
-  if (!cd->javaVM) {
+  auto cd = Instance();
+  if (!cd->java_vm) {
     assert(false);
   }
 
-  return cd->javaVM;
+  return cd->java_vm;
 }
 
 jobject Context::getAndroidContext() {
-  auto cd = instance();
+  auto cd = Instance();
   if (!cd->context) {
     assert(false);
   }
   return cd->context;
 }
-#endif
+#elif defined(__APPLE__)
+void Context::EnterBackground() {
+  auto cd = Instance();
+  std::lock_guard<std::mutex> lock(cd->context_mutex);
+
+  auto& clients = cd->enter_background_subscibers;
+  std::for_each(clients.begin(), clients.end(),
+                [&](std::weak_ptr<EnterBackgroundSubscriber> weak) {
+                  auto strong = weak.lock();
+                  if (strong) {
+                    strong->OnEnterBackground();
+                  }
+                });
+}
+
+void Context::ExitBackground() {
+  auto cd = Instance();
+  std::lock_guard<std::mutex> lock(cd->context_mutex);
+
+  auto& clients = cd->enter_background_subscibers;
+  std::for_each(clients.begin(), clients.end(),
+                [&](std::weak_ptr<EnterBackgroundSubscriber> weak) {
+                  auto strong = weak.lock();
+                  if (strong) {
+                    strong->OnExitBackground();
+                  }
+                });
+}
+
+void Context::StoreBackgroundSessionCompletionHandler(
+    const std::string& sessionName,
+    std::function<void(void)> completionHandler) {
```

Comment:
Variable names are not according to coding style

Reply:
done